fastlane/helper/plugin_scores_helper.rb

```diff
@@ -0,0 +1,184 @@
+          self.value = value
+        end
+      end
+
+      class FastlanePluginScore
+        attr_accessor :name
+        attr_accessor :downloads
+        attr_accessor :info
+        attr_accessor :homepage
+        attr_accessor :raw_hash
+
+        attr_accessor :data
+
+        def initialize(hash)
+          if ENV["FASTLANE_PLUGIN_DIRECTORY_GITHUB_USER"].to_s.length == 0 || ENV["GITHUB_API_TOKEN"].to_s.length == 0
+            raise "Missing ENV variables FASTLANE_PLUGIN_DIRECTORY_GITHUB_USER and/or GITHUB_API_TOKEN"
+          end
+
+          self.name = hash["name"]
+          self.downloads = hash["downloads"]
+          self.info = hash["info"]
+          self.homepage = hash["homepage_uri"] || hash["documentation_uri"]
+          self.raw_hash = hash
+
+          has_github_page = self.homepage.to_s.include?("https://github.com") # TODO: add non GitHub support
+
+          self.data = {
+            has_homepage: self.homepage.to_s.length > 5,
+            has_info: self.info.to_s.length > 5,
+            downloads: self.downloads,
+            has_github_page: has_github_page,
+            mit_license: includes_license?("MIT"),
+            gnu_license: includes_license?("GNU") || includes_license?("GPL"),
+            major_release: Gem::Version.new(hash["version"]) >= Gem::Version.new("1.0.0")
+          }
+
+          if has_github_page
+            self.append_git_data
+            self.append_github_data
+          end
+
+          self.data[:overall_score] = 0
+          self.data[:ratings] = self.ratings.collect do |current_rating|
+            [
+              current_rating.key,
+              current_rating.value
+            ]
+          end.to_h
+
+          self.ratings.each do |current_rating|
+            self.data[:overall_score] += current_rating.value
+          end
+        end
+
+        def ratings
+          [
+            FastlanePluginRating.new(key: :contributors,
+                             description: "The more contributors a project has, the more likely it is it stays alive",
+                                   value: self.data[:github_contributors].to_i * 6),
+            FastlanePluginRating.new(key: :subscribers,
+                             description: "More subscribers = more popular project",
+                                   value: self.data[:github_subscribers].to_i * 3),
+            FastlanePluginRating.new(key: :stars,
+                             description: "More stars = more popular project",
+                                   value: self.data[:github_stars].to_i),
+            FastlanePluginRating.new(key: :forks,
+                             description: "More forks = more people seem to use/modify this project",
+                                   value: self.data[:github_forks].to_i * 5),
+            FastlanePluginRating.new(key: :mit_license,
+                             description: "fastlane is MIT licensed, it's good to have plugins use MIT too",
```

Comment:
I'd say mostly to protect the user. They trust fastlane and it's dependencies is MIT and Apache 2. The plugin generator automatically uses MIT, so if the author of a plugin changes it, there is a reason for it, and it's only fair to also "warn" the user by downgrading the plugin score.